lib/libc/armstdc/include/string.h

```diff
@@ -0,0 +1,19 @@
+#ifndef ZEPHYR_LIB_LIBC_ARMSTDC_INCLUDE_STRING_H_
+#define ZEPHYR_LIB_LIBC_ARMSTDC_INCLUDE_STRING_H_
+
+#include <stddef.h>
+#include <zephyr/toolchain.h>
+
+#include_next <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern size_t strnlen(const char *s, size_t maxlen);
```

Comment:
Is this provided by the library and just not declared in the header? Or is it in a different header?

Also, 'extern' is probably not needed here.

Reply:
This is provided by zephyr itself in `lib/libc/armstdc/src/string.c`.  Need to move that to common in the future.

Reply:
ah, right strnlen isn't a C function, it's a POSIX function. Because of course it is.